### PR TITLE
[RDBMS] PostgreSQL, MySQL migration to GA API

### PR DESCRIPTION
**Pull request by DaeunYim**
**Description**

- PostgreSQL flexible server migration to GA API
- MySQL flexible server migration to GA API
- Add feature for high availability server restart failover
- Change create command default from private scenario to public scenario
- Add tests for network scenarios and validators

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
RDBMS

**Comment by DaeunYim** [comment]
@evelyn-ys  Can I get reviews for this PR? These are mostly migration to GA API and added bunch of tests for client side.

**Comment by evelyn-ys** [comment]
> @evelyn-ys Can I get reviews for this PR? These are mostly migration to GA API and added bunch of tests for client side.

Can you resolve conflicts first?

**Comment by DaeunYim** [comment]
> > @evelyn-ys Can I get reviews for this PR? These are mostly migration to GA API and added bunch of tests for client side.
> 
> Can you resolve conflicts first?

Haha thanks! Just resolved them!

File: src/azure-cli/azure/cli/command_modules/rdbms/_params.py [hunk 1]
```diff
@@ -214,166 +214,249 @@ def _complex_params(command_group):
...
+            arg_type=get_enum_type(['Enabled', 'Disabled']),
+            options_list=['--storage-auto-grow'],
+            help='Enable or disable autogrow of the storage. Default value is Enabled.'
+        )
+
+        yes_arg_type = CLIArgumentType(
+            options_list=['--yes', '-y'],
+            action='store_true',
+            help='Do not prompt for confirmation.'
+        )
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/rdbms/_params.py:311]: There's no need to implement double check arg by yourself.
You can use `confirmation` for command definition. See examples here:
https://github.com/Azure/azure-cli/blob/0850b5f713db57a921e5daf115fbfdffac5f7f31/src/azure-cli/azure/cli/command_modules/storage/commands.py#L108
https://github.com/Azure/azure-cli/blob/0850b5f713db57a921e5daf115fbfdffac5f7f31/src/azure-cli/azure/cli/command_modules/storage/commands.py#L119-L126

**Review by evelyn-ys — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]
Approve for ARM part

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**